internal/condition/user_condition.go

```diff
@@ -0,0 +1,86 @@
+package condition
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/fabric8-services/fabric8-jenkins-idler/internal/model"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+type ProxyResponse struct {
+	Namespace   string `json:"namespace"`
+	Requests    int    `json:"requests"`
+	LastVisit   int64  `json:"last_visit"`
+	LastRequest int64  `json:"last_request"`
+}
+
+// UserCondition covers information about User as provided by the Jenkins Proxy.
+type UserCondition struct {
+	idleAfter time.Duration
+	proxyURL  string
+}
+
+func NewUserCondition(proxyURL string, idleAfter time.Duration) Condition {
+	b := &UserCondition{
+		proxyURL:  proxyURL,
+		idleAfter: idleAfter,
+	}
+	return b
+}
+
+// Eval returns true if there are no buffered request, the last forwarded request occurred more than UserCondition.idleAfter
+// minutes ago and the user accessed the Jenkins UI more than UserCondition.idleAfter minutes ago.
+func (c *UserCondition) Eval(object interface{}) (bool, error) {
+	b, ok := object.(*model.User)
```

Comment:
This should not be pointer to User, but rather just user (other conditions are ok)

```
{"level":"error","msg":"{{{  {    } %!s(int=0)} {New 0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC} {%!s(int=0) {}}} {{  {    } %!s(int=0)} { 0001-01-01 00:00:00 +0000 UTC 0001-01-01 00:00:00 +0000 UTC} {%!s(int=0) {}}} stg2-ebaron1 [{%!s(bool=false) 2018-02-15 14:35:34.034718926 +0000 UTC init} {%!s(bool=true) 2018-02-15 14:36:34.281550291 +0000 UTC Jenkins Unidled for  at 0001-01-01 00:00:00 +0000 UTC}] %!s(int=0) %!s(int=1) 0001-01-01 00:00:00 +0000 UTC 6caa5e71-7c95-40a3-aeb6-cfbd75bf8921} is not of type *User","time":"2018-02-15T15:56:37+01:00"}
```

at the end **is not of type *User**

Reply:
ahh, good catch